drivers/spi/spi_flash.c

```diff
@@ -642,15 +635,15 @@ static void spiflash_sectorerase(FAR struct sim_spiflashdev_s *priv)
   if (priv->wren)
     {
       address = priv->address;
-      if (priv->last_cmd == SPIFLASH_SE)
+      if (priv->last_cmd == SPI_FLASH_SE)
         {
-          address &= CONFIG_SIM_SPIFLASH_SECTORSIZE_MASK;
-          len = CONFIG_SIM_SPIFLASH_SECTORSIZE;
+          address &= CONFIG_SPI_FLASH_SECTORSIZE_MASK;
+          len = CONFIG_SPI_FLASH_SECTORSIZE;
         }
-      else if (priv->last_cmd == SPIFLASH_SSE)
+      else if (priv->last_cmd == SPI_FLASH_SSE)
         {
-          address &= CONFIG_SIM_SPIFLASH_SUBSECTORSIZE_MASK;
-          len = CONFIG_SIM_SPIFLASH_SUBSECTORSIZE;
+          address &= CONFIG_SPI_FLASH_SUBSECTORSIZE_MASK;
+          len = CONFIG_SPI_FLASH_SUBSECTORSIZE;
         }
```

Comment:
```suggestion
        }
      else
        {
          len = 0;
        }

        if (len > 0)
          {
             /*​ Now perform the erase ​*/ 
  
 ​     ​       memset​(&priv->​data​[address], ​0xff​, len);
          }
```

Reply:
Done, but I skip the check before memset since memset will take care this special case.